src/annotator/test/frame-observer-test.js

```diff
@@ -0,0 +1,138 @@
+import FrameObserver, { DEBOUNCE_WAIT } from '../frame-observer';
+
+describe('FrameObserver', () => {
+  let container;
+  let frameObserver;
+  let onFrameAdded;
+  let onFrameRemoved;
+
+  function waitForFrameObserver() {
+    return new Promise(resolve => setTimeout(resolve, DEBOUNCE_WAIT + 10));
+  }
+
+  function createAnnotatableIFrame(
+    attribute = 'enable-annotation',
+    value = ''
+  ) {
+    const frame = document.createElement('iframe');
+    frame.setAttribute(attribute, value);
+    container.appendChild(frame);
+    return frame;
+  }
+
+  function waitForIFrameUnload(frame) {
+    return new Promise(resolve =>
+      frame.contentWindow.addEventListener('unload', resolve)
+    );
+  }
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+    onFrameAdded = sinon.stub();
+    onFrameRemoved = sinon.stub();
+    frameObserver = new FrameObserver(container, onFrameAdded, onFrameRemoved);
+  });
+
+  afterEach(() => {
+    container.remove();
+    frameObserver.disconnect();
+  });
+
+  it('triggers onFrameAdded when an annotatable iframe is added', async () => {
+    let frame = createAnnotatableIFrame();
+    await waitForFrameObserver();
+
+    assert.calledWith(onFrameAdded, frame);
+
+    frame = createAnnotatableIFrame('enable-annotation', 'yes');
+    await waitForFrameObserver();
+
+    assert.calledWith(onFrameAdded, frame);
+
+    frame = createAnnotatableIFrame('enable-annotation', 'true');
+    await waitForFrameObserver();
+
+    assert.calledWith(onFrameAdded, frame);
+
+    frame = createAnnotatableIFrame('enable-annotation', '1');
+    await waitForFrameObserver();
+
+    assert.calledWith(onFrameAdded, frame);
+
+    frame = createAnnotatableIFrame('enable-annotation', 'false'); // the actual value of the attribute is irrelevant
+    await waitForFrameObserver();
+
+    assert.calledWith(onFrameAdded, frame);
+  });
+
+  it("doesn't triggers onFrameAdded when non-annotatable iframes are added", async () => {
+    createAnnotatableIFrame('dummy-attribute');
+    await waitForFrameObserver();
+
+    assert.notCalled(onFrameAdded);
+  });
+
+  it('removal of the annotatable iframe triggers onFrameRemoved', done => {
+    sinon.stub(frameObserver, '_removeFrame').callThrough();
```

Comment:
I am happy to remove this.

The reason why I added this is because there are two paths for executing the `_removeFrame` method. I wanted to check that only one path is taken and the method is only called once.